services/state/addressbook/node.proto

```diff
@@ -0,0 +1,109 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import "basic_types.proto";
+
+option java_package = "com.hederahashgraph.api.proto.java";
+// <<<pbj.java_package = "com.hedera.hapi.node.state.token">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * Representation of a Node in the network Merkle tree
+ *
+ * As with all network entities, a Node has a unique nodeId, which is a long type
+ */
+message Node {
+    /**
+     * The unique id of the Node.
+     */
+    int64 node_id = 1;
+
+    /**
+     * The account is charged for transactions submitted by the node that fail due diligence
+     */
+    AccountID account_id = 2;
+
+    /**
+     * A description of the node, with UTF-8 encoding up to 100 bytes
+     */
+    string description = 3;
+
+    /**
+     * Node Gossip Endpoints, ip address or FQDN and port
+     */
+    repeated ServiceEndpoint gossip_endpoint = 4;
+
+    /**
+     * A node's service Endpoints, ip address or FQDN and port
+     */
+    repeated ServiceEndpoint service_endpoint = 5;
+
+    /**
+     * node’s public key within the Address Book used for signing.
+     * The node's X509 RSA public key used to sign stream files (e.g., record stream
+     * files).
+    */
+    bytes public_key = 6;
+
+    /**
+     * node x509 certificate hash. Hash of the node's TLS certificate. Precisely, this field is a string of
+     * hexadecimal characters which, translated to binary, are the SHA-384 hash of
+     * the UTF-8 NFKD encoding of the node's TLS cert in PEM format. Its value can be
+     * used to verify the node's certificate it presents during TLS negotiations.
+     */
+    bytes certificate_hash = 7;
+
+    /**
+    * The consensus weight of this node in the network.
+    */
+    int64 weight = 8;
+
+    /**
+     * an enum to indicate the status of the node
+     */
+    //    NodeStatus status = 9; not used in phase 2
+
+}
+
+enum NodeStatus {
+    /**
+     * node deleted
+     */
+    DELETED = 0;
+
+    /**
+     * pending to be added
+     */
```

Comment:
Those status is not used for phase 2 and phase 3 will have big changes. Avoid to use address book term as this would be changed too.

Reply:
How do we intend to differentiate the word "node", then?  "node" is used in _many_ different ways, and part of the specification effort is to make clear which "node" we mean in each case.

Reply:
"pending to be added" is not helpful, so we need to specify what this means and how it will be used.  If there is a phase 3 that changes these details, we can specify the correct behavior for phase 2 now and update to match future behavior when that change is made.

Reply:
Updated.